brainzutils/metrics.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import division
+
+import datetime
+from functools import wraps
+
+import six
+from redis import ResponseError
+
+from brainzutils import cache
+
+NAMESPACE_METRICS = "metrics"
+REDIS_MAX_INTEGER = 2**63-1
+STATS_COUNT_DATE_KEY = "date"
+
+_metrics_site_name = None
+
+
+def init(site):
+    global _metrics_site_name
+    _metrics_site_name = site
+
+
+def metrics_init_required(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        if not _metrics_site_name:
+            raise RuntimeError("Metrics module needs to be initialized before use")
+        return f(*args, **kwargs)
+    return decorated
+
+
+@cache.init_required
+@metrics_init_required
+def increment(metric_name, amount=1):
+    """Increment the counter for a metric by a set amount. If incrementing the counter causes it to go over
+    redis' internal counter limit of 2**63-1, the counter is reset to 0.
+    The metric name ``tag`` is reserved and cannot be used.
+
+    Arguments:
+        metric_name: the name of a metric
+        amount: the amount to increase the counter by, must be 0 or greater (default: 1)
+
+    Raises:
+        ValueError if amount is less than 0 or greater than 2**63-1
+        ValueError if the reserved metric name ``tag`` is used
+    """
+
+    if amount < 0:
+        raise ValueError("amount must be positive")
+    if amount > REDIS_MAX_INTEGER:
+        raise ValueError("amount is too large")
+    if metric_name == "tag":
```

Comment:
nit: this could probably be a list, that way adding more reserved metric names would mean just adding it to a list.